Feature/header

CeeyarRbt opened the pull request:
removed node_modules, added changes to the AppBar

CeeyarRbt committed:
testing my branch

CeeyarRbt committed:
Nothing much, just Added two buttons and a dropdown

CeeyarRbt committed:
AppBar, still not final thou. Got a few things to change here and there

CeeyarRbt committed:
Added Account images and color ro the buttons on the right.

jaderabbit commented on chatterbox/src/App.css:
```diff
@@ -0,0 +1,28 @@
+  text-align: center;
+}
+
+.App-logo {
```
Is this in line with the css standard for the rest of the app?

jaderabbit replied:
why did you edit my original comment instead of replying to it?

jaderabbit commented on chatterbox/src/App.js:
```diff
@@ -0,0 +1,18 @@
+    return (
+      <MuiThemeProvider>
+      <MenuAppBar/>
+  </MuiThemeProvider>
```
indentation

CeeyarRbt replied:
got it

jaderabbit commented on chatterbox/src/Main/Header/index.css:
```diff
@@ -0,0 +1,36 @@
+    background-color: #01B9BD;
+
+} 
+button.MuiButtonBase-root-56.MuiButton-root-41.MuiButton-raised-48 {
```
why do you have 48 and what is root 41? 
bad naming
  I didn't name them, I use CSS extension to get access to divs and components for editing them. and that's how it found them.

wscheffer replied:
Is this a Git Submodule component?
  
  I don't understand, Submodule component how?

jaderabbit replied:
Shouldn't it be somethng that is install with npm?

wscheffer replied:
@jaderabbit no I figured it out it's just an issue with the project structure they ran the `create-react-app chatterbox` in the root folder of the app folder instead of the root folder of their projects folder

jaderabbit commented on chatterbox/src/Main/Header/index.css:
```diff
@@ -0,0 +1,36 @@
+	background-color: #01B9BD; 
+		height: 2.44%;
+		color: #FFFFFF;
+		font-family: "Avenir Next";
```
indentation everywhere pls
  fixed

jaderabbit reviewed (changes requested):
You didn't listen to my post. Please read it and apply it to your code

CeeyarRbt committed:
Added required changes, on the PR

CeeyarRbt committed:
fixed indentation, and naming

grootendorste commented on chatterbox/public/index.html:
```diff
@@ -0,0 +1,40 @@
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
```
Technically this comment isn't correct anymore right?

CeeyarRbt replied:
still is, 
  No one is working on the files outside src

grootendorste commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+import { withStyles } from 'material-ui/styles';
+import './index.css';
+import Oval from '../../assets/Oval.png';
+import Oval_2 from '../../assets/Oval_2.png';
```
Hmm, these Oval images seem more like circles. Also watch your casing of the file names. Lower case with dashes for file names are better, but it depends on what you decided for your filenames.

CeeyarRbt replied:
changed Oval to lower case images

CeeyarRbt replied:
changed Oval to lower case images

grootendorste commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+        const open = Boolean(anchorEl);
+
+        return (
+            <div className={classes.root} color="#01B9BD">
```
I feel the color should come from the styles defined up there.

jaderabbit replied:
agreed

CeeyarRbt replied:
Done

grootendorste commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+                                    />
+                                </Button>
+
+                                <Button aria-owns={open ? 'menu-appbar' : null} color="contrast">
```
Can't this (open ? 'menu-appbar' : null) be replaced by with: open && 'menu-appbar'

CeeyarRbt replied:
added the required change. replaced (open ? 'menu-appbar' : null) by open && 'menu-appbar'

grootendorste commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+    }
+}
+
+MenuAppBar.propTypes = {
```
I think you can move the propTypes to the MenuAppBar class and just make it static.

CeeyarRbt replied:
Moved propTypes to the MenuAppBar class and made it to be static

grootendorste commented on chatterbox/src/Main/index.js:
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react';
+import logo from './logo.svg';
```
Do SVGs work on all browsers?

CeeyarRbt replied:
Not all browsers do but some have a basic support for it. what I did was to use .png files instead of the .SVG

grootendorste commented on chatterbox/src/registerServiceWorker.js:
```diff
@@ -0,0 +1,104 @@
+// To learn more about the benefits of this model, read https://goo.gl/KwvDNy.
+// This link also includes instructions on opting out of this behavior.
+
+const isLocalhost = Boolean(
```
I don't think you need to cast these to boolean the whole time, it's javascript, this statement will result in true or false without the Boolean cast.

CeeyarRbt replied:
registerServiceWorker.js is auto generated.

grootendorste commented on chatterbox/src/registerServiceWorker.js:
```diff
@@ -0,0 +1,104 @@
+            const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`;
+
+            if (isLocalhost) {
+                // This is running on localhost. Lets check if a service worker still exists or not.
```
Comments like this should be taken out. Rather explain what is done in this code than saying things like "Let's check..."

grootendorste replied:
But I like that there are comments.

CeeyarRbt replied:
code in the registerServiceWorker.js is not written by us, however the reviews are noted

grootendorste commented on chatterbox/src/registerServiceWorker.js:
```diff
@@ -0,0 +1,104 @@
+                const installingWorker = registration.installing;
+                installingWorker.onstatechange = () => {
+                    if (installingWorker.state === 'installed') {
+                        if (navigator.serviceWorker.controller) {
```
Add a // TODO: to this code, explaining what needs to happen if it's incomplete.

grootendorste commented on chatterbox/src/registerServiceWorker.js:
```diff
@@ -0,0 +1,104 @@
+}
+
+function checkValidServiceWorker(swUrl) {
+    // Check if the service worker can be found. If it can't reload the page.
```
Comments like these are better fit above the function doxygen style e.g. /** ... */

CeeyarRbt replied:
removed the comments inside functions

grootendorste reviewed (changes requested):
Here are some changes I requested.

CeeyarRbt committed:
changed Oval to lower case images

jaderabbit commented on chatterbox/src/Main/Header/index.css:
```diff
@@ -0,0 +1,41 @@
+}
+
+
+/* .AppBarymai div:first-child {  
```
remove pls

CeeyarRbt replied:
removed commented code

jaderabbit reviewed (dismissed):
Address eddie and my comments pls

CeeyarRbt commented:
I clicked edit by mistake and now I just tried to fix it and just deleted the comment. Sorry Jade


Vriendelike Groete / Kind Regards

Retro Rabbit Professional Services (Pty) Ltd

Retro Rabbit 
Siyabonga Ndovela
(ZA) +27 12 348 7577
www.retrorabbit.co.za 

From: Jade Abbott
Sent: Monday, 08 January 2018 15:52
To: RetroRabbit/BlueLagoon
Cc: Siyabonga Nddovela; Author
Subject: Re: [RetroRabbit/BlueLagoon] Feature/header (#10)

@jaderabbit commented on this pull request.

In chatterbox/src/App.css:
> @@ -0,0 +1,28 @@
+.App {
+  text-align: center;
+}
+
+.App-logo {
why did you edit my original comment instead of replying to it?
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

CeeyarRbt commented:
Added the required changes. 

changed Oval to lower case images
changed the color from the styles at the top instead of the divs
removed commented code
Moved propTypes to the MenuAppBar class and made it to be static
replaced (open ? 'menu-appbar' : null) by open && 'menu-appbar'

CeeyarRbt committed:
Added the required changes.

changed Oval to lower case images
changed the color from the styles at the top instead of the divs
removed commented code
Moved propTypes to the MenuAppBar class and made it to be static
replaced (open ? 'menu-appbar' : null) by open && 'menu-appbar'

KimberleighBeyl commented on chatterbox/.gitignore:
```diff
@@ -0,0 +1,21 @@
+# See https://help.github.com/ignore-files/ for more about ignoring files.
+
```
Why 2 .gitignores?

CeeyarRbt replied:
It's the project structure. we created chatterbox inside root folder but the one outside doesn't work and will be deleted.

KimberleighBeyl replied:
ok

KimberleighBeyl commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+    };
+
+    handleChange = (event, checked) => {
+        this.setState({ auth: checked });
```
Change to redux

KimberleighBeyl commented on chatterbox/src/Main/Header/index.js:
```diff
@@ -0,0 +1,132 @@
+    };
+
+    handleMenu = event => {
+        this.setState({ anchorEl: event.currentTarget });
```
Same as previous comment, no setstate, must be redux

CeeyarRbt replied:
That's the current plan now.

CeeyarRbt committed:
removed the 2nd .gitIgnore

resoved conflics

CeeyarRbt committed:
Added gitignore in the correct directory

CeeyarRbt committed:
Updated package.json

CeeyarRbt committed:
Added README to correct directory

CeeyarRbt committed:
added nackground color

KimberleighBeyl reviewed (changes requested):
Merge conflicts

CeeyarRbt commented:
Working on the.


Vriendelike Groete / Kind Regards

Retro Rabbit Professional Services (Pty) Ltd

Retro Rabbit 
Siyabonga Ndovela
(ZA) +27 12 348 7577
www.retrorabbit.co.za 

From: KimberleighBeyl
Sent: Tuesday, 09 January 2018 11:49
To: RetroRabbit/BlueLagoon
Cc: Siyabonga Nddovela; Author
Subject: Re: [RetroRabbit/BlueLagoon] Feature/header (#10)

@KimberleighBeyl requested changes on this pull request.
Merge conflicts
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

CeeyarRbt committed:
Merge branch 'develop' into feature/header

# Conflicts:
#	chatterbox/package-lock.json
#	chatterbox/package.json
#	chatterbox/public/index.html
#	chatterbox/public/manifest.json
#	chatterbox/src/App.js
#	chatterbox/src/Auth/Login/index.js
#	chatterbox/src/Auth/Register/index.js
#	chatterbox/src/Auth/index.js
#	chatterbox/src/Main/index.js
#	chatterbox/src/index.js

CeeyarRbt committed:
fixed issues after merge

CeeyarRbt committed:
Merge branch 'develop' into feature/header

# Conflicts:
#	chatterbox/src/App.js
#	chatterbox/src/Main/MainArea/ChattingComponent/Sidebar/index.js

CeeyarRbt committed:
Resolved conflicts
